src/components/MissionItem.js

```diff
@@ -1,32 +1,39 @@
 import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
-import { missionJoined } from '../redux/features/missionSlice';
+import { missionJoined, missionLeaved } from '../redux/features/missionSlice';
 
 const MissionItem = ({ mission }) => {
   const dispatch = useDispatch();
 
   const {
     mission_name, mission_id, description, reserved,
   } = mission;
-  const [membership, actionable] = (
-    reserved
-      ? ['Active Member', 'Leave Mission']
-      : ['NOT A MEMBER', 'Join Mission']
-  );
 
-  const onClick = () => {
+  const joinMissionHandler = () => {
     dispatch(missionJoined(mission_id));
   };
 
+  const LeaveMissionHandler = () => {
+    dispatch(missionLeaved(mission_id));
+  };
+
   return (
     <tr>
       <td>{mission_name}</td>
       <td>{description}</td>
       <td>
-        <span>{membership}</span>
+        {reserved ? <span>Active Member</span> : <span>Leave Mission</span>}
```

Comment:
When `reserved = false`, your badge text should be `NOT A MEMBER` as per the original project design.
Kindly refer to the project screenshot to confirm.